pkg/agent/controller/packetcapture/bpf/parser.go

```diff
@@ -0,0 +1,154 @@
+package bpf
+
+import (
+	"encoding/binary"
+	"strings"
+
+	"golang.org/x/net/bpf"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	crdv1alpha1 "antrea.io/antrea/pkg/apis/crd/v1alpha1"
+	binding "antrea.io/antrea/pkg/ovs/openflow"
+)
+
+func GetRawInstruction(insts []bpf.Instruction) ([]bpf.RawInstruction, error) {
+	return bpf.Assemble(insts)
+
+}
+
+// CompilePacketFilter compile our crd spec to bpf instructions. For now, we only focus on
+// ipv4 traffic. Compare to raw BPF filter supported by libpcap, we only need to support
+// limited usercases so an expression parser is not needed.
+func CompilePacketFilter(spec *crdv1alpha1.PacketCaptureSpec, matchPacket *binding.Packet) []bpf.Instruction {
```

Comment:
I find it quite confusing that we need both the `spec.Packet` and the `binding.Packet` as inputs to this function. It feels like we have 2 different sources of truth. Could you explain why?

Reply:
in the `spec.Packet`, Pod information need to be parsed as `IP` to be compiled to BPF instructions. binding.Packet saved the pared ip information for that.  Parse pod ip need the `Controller` instance, so a `binding.Packet` can simply the workflow. The interface does look confusing. I can add more comments,  also will see if there be some better approches.

Reply:
removed this due to the interface change.